tests/js/client/shell/api/rest-schema-handler.js

```diff
@@ -0,0 +1,628 @@
+
+            const descView = body.views.find(v => v.viewName === 'descView');
+            assertTrue(descView, 'descView should exist');
+            assertEqual(descView.links.length, 2, 'descView should have 2 links');
+        },
+
+        test_InvalidSampleNumValues_ShouldReturn400: function () {
+            [
+                '?sampleNum=-1',
+                '?sampleNum=12.3',
+                '?sampleNum=abc',
+                '?sampleNum=999999999999999999999999999999'
+            ].forEach(param => {
+                const doc = arango.GET_RAW(api + param);
+                assertEqual(400, doc.code, `Expected HTTP 400 for invalid sampleNum '${param}'`);
+                assertTrue(doc.parsedBody.error, 'Error flag should be true for invalid sampleNum');
+            });
+        },
+
+        test_InvalidExampleNumValues_ShouldReturn400: function () {
+            [
+                '?exampleNum=-1',
+                '?exampleNum=12.3',
+                '?exampleNum=abc',
+                '?exampleNum=999999999999999999999999999999'
+            ].forEach(param => {
+                const doc = arango.GET_RAW(api + param);
+                assertEqual(400, doc.code, `Expected HTTP 400 for invalid exampleNum '${param}'`);
+                assertTrue(doc.parsedBody.error, 'Error flag should be true for invalid exampleNum');
+            });
+        },
+
+        test_EmptySampleNum_ShouldReturn200: function () {
+            const doc = arango.GET_RAW(api + '?sampleNum=');
+            assertEqual(200, doc.code, 'Empty sampleNum should default to valid response');
+        },
+
+        test_EmptyExampleNum_ShouldReturn200: function () {
+            const doc = arango.GET_RAW(api + '?exmapleNum=');
+            assertEqual(200, doc.code, 'Empty exampleNum should default to valid response');
+        },
+
+        test_ExampleNumGreaterThanSampleNum_ShouldReturn400: function () {
+            const doc = arango.GET_RAW(api + '?sampleNum=10&exampleNum=20');
+            assertEqual(400, doc.code, 'exampleNum greater than sampleNum should return 400');
+            assertTrue(doc.parsedBody.error, 'Error flag should be true when exampleNum > sampleNum');
+        },
+
+        test_ValidExampleNum_ShouldReturnCorrectExampleCount: function () {
+            const doc = arango.GET_RAW(api + '?sampleNum=10&exampleNum=3');
+            const products = doc.parsedBody.collections.find(c => c.collectionName === 'products');
+            assertEqual(200, doc.code, 'Valid exampleNum should return HTTP 200');
+            assertTrue(products, 'products collection should be present');
+            assertEqual(
+                products.examples.length,
+                3,
+                'Expected exactly 3 examples for products when exampleNum=3'
+            );
+        },
+
+        test_GetSchemaWithSampleNumOnly_ShouldReturnSchemasWithNoOptionalAttributes: function () {
```

Comment:
I do not See a `test_GetCollectionByName_ShouldReturnCollectionSchema`

Also I would like you to add tests for the Indexes (I know that is still on your List)